src/Backend/Modules/ContentBlocks/Domain/ContentBlock/ContentBlock.php

```diff
@@ -125,26 +124,29 @@ private function __construct(
         $this->status = $status;
     }
 
-    public static function create(
-        int $id,
-        int $userId,
-        int $extraId,
-        Locale $locale,
-        string $title,
-        string $text,
-        bool $isHidden,
-        string $template = self::DEFAULT_TEMPLATE
-    ) : self {
+    public static function fromDataTransferObject(ContentBlockDataTransferObject $dataTransferObject)
+    {
+        if ($dataTransferObject->hasExistingContentBlock()) {
+            $dataTransferObject->getContentBlockEntity()->status = Status::archived();
+
+            return self::create($dataTransferObject);
```

Comment:
you repeat what is done below so you can just have the ifstatement and use one return here